packages/lib/sdk/src/usql/query/Query.js

```diff
@@ -367,8 +367,12 @@ ${this.text.trim()}
 		);
 		return resolved;
 	};
-	fetch = async () => {
-		return Promise.allSettled([this.#fetchColumns(), this.#fetchData()]).then(() => this.value);
+	fetch = () => {
+		const cols = this.#fetchColumns();
+		if (cols instanceof Promise) {
+			return Promise.allSettled([this.#fetchColumns(), this.#fetchData()]).then(() => this.value);
+		}
+		this.#fetchData();
```

Comment:
```suggestion
		this.#fetchData();
		return Promise.resolve();
```
I think we should return `Promise.resolve()` so people can assume the return value is thenable

Reply:
This means that `instanceof Promise` will always be true though, so the current way we check for async code could no longer wrok